### PR TITLE
[compiler] Support outlined fns for auto-deps

### PR DESCRIPTION
**Pull request by jbrown215**
Summary:

We can keep track of outlined functions and insert an empty dep array when we find one passed to an effect. Outlined functions
always have no deps!

Outlined functions are given unique names, so we can look at LoadGlobals and see if one of those names is used.
---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/facebook/react/pull/31662).
* __->__ #31662
* #31664
* #31663

### CODE REVIEW

**Review by mofeiZ — APPROVED** [review]
shipit!

File: compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/outlined-function.expect.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ import { print } from "shared-runtime";
  * before OutlineFunctions
  */
 function OutlinedFunctionInEffect() {
-  useEffect(_temp);
+  useEffect(_temp, []);
```
Comment (mofeiZ) [inline compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/outlined-function.expect.md:37]: nice!

**Comment by josephsavona** [comment]
I don't get it - shouldn't outlined functions already be a LoadGlobal, which doesn't need to be a dependency?

**Comment by jbrown215** [comment]
> I don't get it - shouldn't outlined functions already be a LoadGlobal, which doesn't need to be a dependency?

hmm, so IIUC we don't need to keep track of outlined functions because it's sufficient to check if the argument is a LoadGlobal? Sounds reasonable to me

**Comment by josephsavona** [comment]
Yeah exactly

**Pull request closed without merging**